libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Extensions/RestClient.swift

```diff
@@ -179,4 +266,43 @@ extension RestClient {
             }
         }
     }
+  
+    // MARK: Record Convience API - Swift & Combine
+
+  
+    /// This method provides a reusuable, generic Combine pipeline for retrieving records
+    ///   from Salesforce. It relys on Swift Generics, and type inference to determine what
+    ///  models to create.
+    ///
+    /// Given a model object - Contact, you can use this pipeline like this:
+    /// contactsForCancellable = RestClient.shared.records(forRequest: request)
+    ///   .receive(on: RunLoop.main)
+    ///   .assign(to: \.contacts, on: self)
+    ///
+    /// This pipeline infers it's return type from the variable in the assign subscriber.
+    func records<Record: Decodable>(forRequest request:RestRequest ) -> AnyPublisher<[Record], Never> {
+      guard request.isQueryRequest else {
+        return Empty(completeImmediately: true).eraseToAnyPublisher()
+      }
+      return RestClient.shared.publisher(for: request)
+        .tryMap({ (response) -> Data in
+          response.asData()
+        })
+        .decode(type: QueryResponseWrapper<Record>.self, decoder: JSONDecoder())
+        .map({ (record) -> [Record] in
+          record.records
+        })
+        .catch({ _ in
+          Just([Record]())
+        })
+        .eraseToAnyPublisher()
+    }
+  
+    /// Reusable, generic Combine Pipeline returning an array of records of a local
+    /// model object that conforms to Decodable. This method accepts a query string and defers
+    /// to records<Record:Decodable>(forRequest request: RestRequest) -> AnyPublisher<[Record], Never>
+    func records<Record: Decodable>(forQuery query:String, withApiVersion version: String = SFRestDefaultAPIVersion ) -> AnyPublisher<[Record], Never> {
```

Comment:
@codefriar Return QueryResponse instead?

Reply:
Yep, taken care of.